From 54df39c8e53cff6fec977c1830dc96e9f6e49ae4 Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Fri, 14 Dec 2012 17:04:58 +0800
Subject: [PORT FROM MAIN] usb3/adb-host: fix usb3.0 adb enumeration failure on Linux host

BZ: 41584

This patch modifies adb user space source code compiled for Linux
host side executable bin file, in order to support super speed adb
device enumeration.

The Macro defined in ch9.h is not used right now.

Change-Id: I6950576400b1ff4d881a72ab04262bede0bc57bf
Orig-Change-Id: I3a99aa34d4d58bfc45a864aa9cdc5dba1223d793
Category: aosp improvement
Domain: USB ADB
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
Reviewed-on: http://android.intel.com:8080/81699
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gao, Yunpeng <yunpeng.gao@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Wang, Ran <ran.wang@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 adb/usb_linux.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/adb/usb_linux.c b/adb/usb_linux.c
index 7bf2057..6a5cb0a 100644
--- a/adb/usb_linux.c
+++ b/adb/usb_linux.c
@@ -237,6 +237,15 @@ static void find_usb_device(const char *base,
                             // looks like ADB...
                         ep1 = (struct usb_endpoint_descriptor *)bufptr;
                         bufptr += USB_DT_ENDPOINT_SIZE;
+                        /* USB3.0 endpoint companion descriptor:
+                         * set USB_DT_SS_ENDPOINT_COMP and
+                         * USB_DT_SS_EP_COMP_SIZE as fixed value for now,
+                         * as Android header file has not been updated yet.
+                         */
+                        if (bufptr[0] == 0x06 &&
+                            bufptr[1] == 0x30)
+                            bufptr += 6;
+
                         ep2 = (struct usb_endpoint_descriptor *)bufptr;
                         bufptr += USB_DT_ENDPOINT_SIZE;
 
-- 
1.7.4.1

